README.md

```diff
@@ -1,22 +1,91 @@
+version of the package.
+
+# Reproducing Results
+
+## Training adversarial policies
+
+`modelfree.train` trains a single adversarial policy. `modelfree.multi.train` trains multiple
+adversarial policies, using Ray (see below) for parallelization.
+
+To replicate the results in the paper (there may be slight differences due to randomness not
+captured in the seeding), run `python -m modelfree.multi.train with paper`. To run the
+hyperparameter sweep, run `python -m modelfree.multi.train with hyper`.
+
+You can find results from our training run on s3://adversarial-policies-public/multi_train/paper.
+This includes TensorBoard logs, final model weights, checkpoints, and individual policy configs.
+Run `experiments/pull_public_s3.sh` to sync this and other data to `data/aws-public/`.
+
+## Evaluating adversarial policies
+
+`modelfree.score_agent` evaluates a pair of policies, for example an adversary and a victim.
+It outputs the win rate for each agent and the number of ties. It can also render to the screen
+or produce videos.
+
+We similarly use `modelfree.multi.score` to evaluate multiple pairs of policies in parallel.
+To reproduce all the evaluations used in the paper, run the following bash scripts, which call
+`modelfree.multi.score` internally:
+  - `experiments/modelfree/baselines.sh`: fixed baselines (no adversarial policies).
+  - `experiments/modelfree/attack_transfers.sh <path-to-trained-adversaries>`. To use our
+     pre-trained policies, use the path `data/aws-public/multi_train/paper/20190429_011349`.
+
+## Visualizing Results
+
+Most of the visualization code lives in the `modelfree.visualize` package. To reproduce the figures
+in the paper, use `paper_config`; for those in the appendix, use `supplementary_config`. So:
+
+```bash
+  python -m modelfree.visualize.scores with paper_config  # heatmaps in the paper
+  python -m modelfree.visualize.training with supplementary_config  # training curves in appendix
+```
+
+To re-generate all the videos, use `modelfree.visualize.make_videos`. We would recommend running
+in Docker using `Xdummy` for this: there are rendering issues with many graphics drivers.
+
+Note you will likely need to change the default paths in the config to point at your experimental
+results and desired output directory.
+
+## Additional Analysis
+
+The density modeling can be run by `experiments/modelfree/density.sh`, or with custom
+configurations via `modelfree.density.pipeline`.
+
+The t-SNE visualizations can be replicated with `modelfree.tsne.pipeline`.
+
+## Using Ray
+
+Many of the experiments are computationally intensive. You can run them on a single machine, but it
+might take several weeks. We use [Ray](https://github.com/ray-project/ray) to run distributed
+experiments. We include example configs in `src/modelfree/configs/ray/`. You will need to, at a
+minimum, edit the config to use your own AMI (anything with Docker should work) and private key.
```

Comment:
It seems we're assuming they're editing the aws.yaml config, which we may want to make explicit

Reply:
Thanks, made this explicit.